acceptance/api_v1_services_mutation_test.go

```diff
@@ -0,0 +1,787 @@
+			service = newServiceName()
+		})
+
+		It("returns a 'bad request' for a non JSON body", func() {
+			response, err := Curl("DELETE",
+				fmt.Sprintf("%s/api/v1/orgs/idontexist/services/%s",
+					serverURL, service),
+				strings.NewReader(""))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusBadRequest), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("unexpected end of JSON input\n"))
+		})
+
+		It("returns a 'bad request' for a non-object JSON body", func() {
+			response, err := Curl("DELETE",
+				fmt.Sprintf("%s/api/v1/orgs/idontexist/services/%s",
+					serverURL, service),
+				strings.NewReader(`[]`))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusBadRequest), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("json: cannot unmarshal array into Go value of type models.DeleteRequest\n"))
+		})
+
+		It("returns a 'not found' when the org does not exist", func() {
+			response, err := Curl("DELETE",
+				fmt.Sprintf("%s/api/v1/orgs/idontexist/services/%s",
+					serverURL, service),
+				strings.NewReader(`{ "unbind": false }`))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("Organization 'idontexist' does not exist\n"))
+		})
+
+		It("returns a 'not found' when the service does not exist", func() {
+			response, err := Curl("DELETE",
+				fmt.Sprintf("%s/api/v1/orgs/%s/services/bogus", serverURL, org),
+				strings.NewReader(`{ "unbind": false }`))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("service 'bogus' not found\n"))
+		})
+
+		Context("With bound applications", func() {
+			var app string
+			var service string
+
+			BeforeEach(func() {
+				service = newServiceName()
+				app = newAppName()
+				makeCustomService(service)
+				makeApp(app)
+				bindAppService(app, service, org)
+			})
+
+			AfterEach(func() {
+				cleanupApp(app)
+				cleanupService(service)
+			})
+
+			It("returns 'bad request'", func() {
+				response, err := Curl("DELETE",
+					fmt.Sprintf("%s/api/v1/orgs/%s/services/%s",
+						serverURL, org, service),
+					strings.NewReader(`{ "unbind": false }`))
```

Comment:
Can we make `"unbind": false` the default? Maybe we can also allow an empty body too. E.g. if body is an empty string we don't even try to parse it. If it's not empty then it should be JSON.

Reply:
Hm. It feels like a false optimization to me at the moment. What we have here is trivial to document, and trivial to implement cli side. Adding the exemptions makes the docs more complex at least. We could keep the cli side unchanged, i.e. simple, but then the exemption is just ballast server side.

In short, I do not see the value here, only the problems.

So, coming down on the no side here.

Reply:
ok agreed.